### ISSUE TITLE
Automatically increment version numbers for hotfixes?

### ISSUE THREAD

**Issue by roji** [GitTools/GitVersion#634]
I imagine this question was raised before, sorry for the duplication if so.

I used continuous deployment in project to deliver unstable nugets to users. This is working very well, except where hotfix versions are concerned. In other words, all commits in branch `hotfix/3.0.2` get assigned the nuget version `3.0.2-beta0001`. This makes it very difficult to deliver packages for testing as the hotfix is being committed to.

Is there a reason not to employ the same strategy as for developer, i.e. simply call it `3.0.2-unstable0004`, incrementing the number with commits?

**Comment by JakeGinnivan** [comment]
The hotfix branch works just like the release branch. You can change the configuration to use continuous deployment mode if you want.

The reason we do not use the same strategy as develop is that the continuous deployment mode makes it very hard to follow SemVer directly as you end up skipping multiple pre-release tags each release. SemVer should really only change a single value on each *release*, but seeing as you never release from develop then we set that to continuous deployment by default.

Hope that answers the question

**Comment by roji** [comment]
Thanks for the quick info!

Interesting, when using gitflow I was under the impression that the hotfix branch is more similar to the *develop* branch. After all, you don't release from hotfix - you merge it into master and release from there.

If it's not too much trouble can you provide a scenario illustrating the SemVer incompatibility when using continuous deployment for hotfix?

**Comment by JakeGinnivan** [comment]
Yep sure.

```
hotfix/3.0.2  -->
3.0.2-beta.1
*commit*
3.0.2-beta.2
*release 3.0.2-beta.2*
```

where is beta.1? Oh it was transient and never got released.

**Comment by JakeGinnivan** [comment]
For that reason I separate builds and deployments. Builds produce the same SemVer until I release a build. At that point I tag the git repo and then the following commit will move on

**Comment by JakeGinnivan** [comment]
We are looking at a new way to solve this issue at #621 - I am talking with the Octopus guys to see if that could be a good way forward to solve this issue.

**Comment by roji** [comment]
Thanks, yes, that issue pretty much sums up what I'd want.

My (not necessarily correct) understanding was that doing beta *releases* was something restricted to minor versions, not hotfixes. If you need to support beta (or other prerelease) releases of hotfixes then I definitely see your point - will look forward to it being solved.

Meanwhile I'll try to enable continuous deployment for hotfix on my project. Thanks!

**Comment by JakeGinnivan** [comment]
>  If you need to support beta (or other prerelease) releases of hotfixes then I definitely see your point - will look forward to it being solved.

It is supported now, you just tag the commit when you release it, and the pre-release tag will increment
